ldap/servers/slapd/back-ldbm/ldbm_entryrdn.c

```diff
@@ -1032,6 +1033,20 @@ entryrdn_lookup_dn(backend *be,
     dblayer_value_free(be, &data);
     dblayer_value_init(be, &data);
 
+    /* Just in case the suffix ID is not '1' retrieve it from the database */
+    keybuf = slapi_ch_strdup(slapi_sdn_get_ndn(be->be_suffix));
+    dblayer_value_set(be, &key, keybuf, strlen(keybuf) + 1);
+    rc = dblayer_cursor_op(&ctx.cursor, DBI_OP_MOVE_TO_KEY, &key, &data);
+    if (rc) {
+        slapi_log_err(SLAPI_LOG_WARNING, "entryrdn_lookup_dn",
+                  "Fails to retrieve the ID of suffix %s - Assume it is '1'\n", slapi_sdn_get_ndn(be->be_suffix));
+        suffix_id = 1;
+    } else {
+        elem = (rdn_elem *)data.data;
+        suffix_id = id_stored_to_internal(elem->rdn_elem_id);
+        dblayer_value_free(be, &data);
+    }
+
```

Comment:
Memory leak: I think that both
```
dblayer_value_free(be, &data);
dblayer_value_free(be, &key);
```
are needed (whether rc==0 or not)

Reply:
Nice catch. Thanks.
I updated the fix